src/ass/kfx.py

```diff
@@ -0,0 +1,237 @@
+) -> Sequence[pyass.Tag]:
+    if kChar.karaDuration == timedelta():
+        return []
+    # Karaoke timing
+    if len(resultTag) == 0:
+        return [pyass.KaraokeTag(kChar.karaDuration)]
+    else:
+        return [
+            pyass.TransformTag(start=kChar.karaStart, end=kChar.karaEnd, to=resultTag)
+        ]
+
+
+def to_default_event(
+    line: KLine,
+    actorToStyle: Mapping[str, Sequence[pyass.Tag]],
+    switchDuration: timedelta,
+    transitionDuration: timedelta,
+) -> pyass.Event:
+    if line.isEN:
+        line.calculate_char_offsets(EN_STYLE, transitionDuration)
+    else:
+        line.calculate_char_offsets(ROMAJI_STYLE, transitionDuration)
+
+    # Generate line style tags
+    eventParts: list[pyass.EventPart] = [
+        pyass.EventPart(
+            tags=[
+                *LYRICS_TAGS,
+                get_en_pos_tag(line) if line.isEN else get_romaji_pos_tag(line),
+            ],
+        )
+    ]
+
+    # Set color if there's a constant actor
+    if len(line.actorSwitches) == 0:
+        eventParts.append(pyass.EventPart(tags=actorToStyle[line.startActor]))
+
+    # Leading switch duration
+    if not line.isEN:
+        eventParts.append(pyass.EventPart(tags=[pyass.KaraokeTag(switchDuration)]))
+
+    # Generate character style tags
+    for char in line.chars:
+        eventParts.append(
+            pyass.EventPart(
+                tags=[
+                    *get_char_transform_tags(char, switchDuration, transitionDuration),
+                    *get_char_actor_tag(char, actorToStyle, switchDuration),
+                    *get_char_karaoke_tag(char),
+                ],
+                text=char.char,
+            )
+        )
+
+    # Trailing switch duration
+    if not line.isEN:
+        eventParts.append(pyass.EventPart(tags=[pyass.KaraokeTag(switchDuration)]))
+
+    return pyass.Event(
+        format=get_line_format(line),
+        style=EN_STYLE.name if line.isEN else ROMAJI_STYLE.name,
+        start=line.start - switchDuration,
+        end=line.end + switchDuration,
+        parts=eventParts,
+    )
+
+
+def get_char_syl_karaoke_tag(
+    kChar: KChar, switchDuration: timedelta, resultTag: Sequence[pyass.Tag] = []
+) -> Sequence[pyass.Tag]:
+    if kChar.karaDuration == timedelta():
+        return []
+    # Karaoke timing
+    if len(resultTag) == 0:
+        return [pyass.KaraokeTag(kChar.karaDuration)]
+    else:
+        return [
+            pyass.TransformTag(
+                start=kChar.syl.start + switchDuration,
+                end=kChar.syl.end + switchDuration,
+                to=resultTag,
+            )
+        ]
+
+
+def to_shad_event(
+    line: KLine,
+    actorToStyle: Mapping[str, Sequence[pyass.Tag]],
+    switchDuration: timedelta,
+    transitionDuration: timedelta,
+) -> pyass.Event:
+    line.calculate_char_offsets(ROMAJI_STYLE)
+    firstColor = Color.parse("&HBBC9F8&")
+    switchColor = Color.parse("&H8A78D7&")
```

Comment:
Instead of constructing the color from parsing, can instead construct it using the constructor (which has the benefit of making it slightly easier to read than parsing strings directly:


```python
firstColor = Color(r="0x...", g="0x...", b="0x...") ...
```

Reply:
Yeah, I could do that, but I copied the colors from someone else's .ass file anyway, so it was way easier to just do it like this.